src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentSynchronizationEndpoint.cs

```diff
@@ -174,5 +169,10 @@ internal SourceText ApplyContentChanges(IEnumerable<TextDocumentContentChangeEve
 
             return sourceText;
         }
+
+        public TextDocumentAttributes GetTextDocumentAttributes(DocumentUri uri)
```

Comment:
@david-driscoll what are text document attributes anyhow?

Reply:
I had a small note in the docs I'm starting on: https://github.com/OmniSharp/csharp-language-server-protocol/pull/338

They are used to route documents to the correct handlers.

If you had two completion providers with `DocumentSelectors`, one that serves `visualbasic` and another that serves `csharp`, You would want to look at the extension `*.vb` -> `visualbasic` / `*.cs` => `csharp` and set the correct attribute.

If you wanted to have different completion handlers for `*.cshtml` vs `*.razor` for example you could do it through the `DocumentSelector`

Reply:
Ahhh very interesting, thanks for the explanation!